constants.py

```diff
@@ -1 +1,179 @@
+import math
 
+from robotpy_toolkit_7407.utils.units import hour, m, mile, rad, rev, s
+from wpimath.geometry import Pose3d, Rotation3d, Transform3d
+
+from config import field_length, field_width
+from units.SI import (
+    inches_to_meters,
+    meters,
+    meters_per_second,
+    meters_per_second_squared,
+    radians,
+    radians_per_second,
+    rotations,
+    rotations_per_minute,
+    rotations_per_minute_per_second,
+)
+
+#TODO Change Camera Positions
+#TODO Change drivetrain gear ratios
+#TODO Change drivetrain track width and robot length
+#TODO update drivetrain wheel speeds
+
+period = 0.03
+
+# --- DRIVETRAIN ---
+
+drivetrain_turn_gear_ratio: rotations = 150 / 7  # 21.428
+drivetrain_move_gear_ratio_as_rotations_per_meter = 21.148
+
+drivetrain_move_gear_ratio: rotations_per_minute = (
+    drivetrain_move_gear_ratio_as_rotations_per_meter * 60
+)  # 20.64 * 62
+
+track_width: meters = 0.50156
+robot_length: meters = 0.635
```

Comment:
you can get most of these from the CAD, if not messaging Cam or Huber should do the trick